Fixed debian dependencies

mikhailnov opened the pull request:
* improved dependencies
* improved installation instructions
* added Ubuntu 18.10 cosmic to the PPA building script

Moved dracut-core, smartmontools to 'Suggests', inxi and numactl do not seem to have odd dependencies, so I left them in 'Recommends'

lvc commented on INSTALL:
```diff
@@ -38,7 +38,6 @@ Content:
    - smartmontools
    - pciutils
    - usbutils
-   - edid-decode
```
It's required to collect EDIDs. Why moving to recommended?

lvc commented on debian/control:
```diff
@@ -8,8 +8,65 @@ Homepage: https://github.com/linuxhw/hw-probe
 
 Package: hw-probe
 Architecture: all
-Depends: perl-base (>=5), libdigest-sha-perl, curl, findutils, hwinfo, dmidecode, pciutils, usbutils, hdparm, smartmontools, edid-decode, util-linux, lsb-release, fdisk | util-linux, coreutils
```
Need also smartmontools log in all probes.

mikhailnov commented:
Fixed.
Smartmontools 6.6 with fixed dependencies has been built in the same PPA: https://launchpad.net/~mikhailnov/+archive/ubuntu/hw-probe/+packages

Also updated docs.

linuxhw commented on README.md:
```diff
@@ -28,39 +27,37 @@ You can make a probe with the help of a script, Docker image or Linux ISO (see b
+## Usage
 
-    sudo hw-probe -all -upload -id DESC
+`sudo hw-probe -all -upload -id DESC`
```
This is not readable outside Github markdown viewer, so let's leave formatting for code as is.

mikhailnov replied:
Да вроде нормально читается в любом редакторе с минималньой подсветкой синтаксиса. Тут уже и конфликты автомержа подъехали, проще вам будет руками смержить, как нравится,а  не мне править.

linuxhw commented on README.md:
```diff
@@ -28,39 +27,37 @@ You can make a probe with the help of a script, Docker image or Linux ISO (see b
 
-Live ISO
---------
+## Live ISO
```
Let's leave '----' for h2 and '####' for h3.

linuxhw commented on README.md:
````diff
@@ -69,25 +66,63 @@ The tool uploads SHA512 hash of MAC addresses and serial numbers to properly
+sudo apt install hw-probe smartmontools
+```
+   
+If you want only a minimal set of dependencies to be installed, install as:
````
We need to install minimal set of deps by default.

linuxhw commented:
Please squash all commits into one before creating pull requests. See https://help.github.com/articles/about-pull-request-merges/.

lvc commented:
Cherry-picked/pushed effective commits from the request. Can't merge all due to the following commits:

* ce01ef8a2f498a7329449035a4ac6209f37c4819 (isn't based on master)
* 25753849259eabf6d88dea00122baae252c60c4f (wtf?)

Thank you.

mikhailnov commented:
https://github.com/linuxhw/hw-probe/commit/25753849259eabf6d88dea00122baae252c60c4f is a merge-commit after I did:
```
git remote add upstream https://github.com/linuxhw/hw-probe.git
git fetch upstream
git merge upstream/master
```
to sync my fork mikhailnov/hw-probe with linuxhw/hw-probe. I don't know how to avoid such merge commits.
